lvmd/device_class_manager.go

```diff
@@ -72,11 +91,34 @@ func ValidateDeviceClasses(deviceClasses []*DeviceClass) error {
 		if dcNames[dc.Name] {
 			return fmt.Errorf("duplicate device-class name: %s", dc.Name)
 		}
-		if vgNames[dc.VolumeGroup] {
-			return fmt.Errorf("duplicate volume group name: %s, %s", dc.Name, dc.VolumeGroup)
+
+		name := dc.VolumeGroup
+
+		// thinpool validation, ignore any thinpoolconfig if Type is not TypeThin
+		if dc.Type == TypeThin {
+
+			if dc.ThinPoolConfig == nil {
+				return fmt.Errorf("device class type is thin but thinpool config is empty: %s", dc.Name)
+			}
+
+			if len(dc.ThinPoolConfig.Name) == 0 {
+				return fmt.Errorf("thinpool name should not be empty: %s", dc.Name)
+			}
+
+			if dc.ThinPoolConfig.OverprovisionRatio == nil || *dc.ThinPoolConfig.OverprovisionRatio < 1.0 {
+				return fmt.Errorf("overprovision for thin pool %s in device class %s should be greater than 1.0", dc.ThinPoolConfig.Name, dc.Name)
+			}
+			// combination of volumegroup and thinpool should be unique across device classes
+			// so the key 'name' shouldn't appear twice to verify it's uniqueness
+			name = name + "/" + dc.ThinPoolConfig.Name
+		}
```

Comment:
Could you please add a check for dc.Type. It must be `TypeThin`, `TypeThick`, or empty.